### PR TITLE
DEVX-2258: Added article about app builder events auto subscription

### PR DESCRIPTION
**Pull request by kokoc**
## Description

Added article about app builder auto subscriptions to events

## Related Issue
https://jira.corp.adobe.com/browse/DEVX-2258

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by rperelma** [comment]
LGTM, but would like @sgera to also review (and merge)

File: gatsby-config.js [hunk 1]
```diff
@@ -51,6 +51,10 @@ module.exports = {
             title: "Introduction to Journaling",
             path: "guides/journaling_intro.md",
           },
+          {
+            title: "App Builder Auto Subscriptions",
+            path: "guides/autosubscriptions.md",
+          },
```
Comment (sgera) [inline gatsby-config.js:57]: I think we should make this a child page of `Runtime Actions As Webhook` (guides/runtime_webhooks.md) as this is relevant for `App Builder / Runtime actions` integrations only. wdyt?

File: src/pages/guides/autosubscriptions.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+## Creating self-contained application
+In this chapter, we will create a code that listens to a specific event type and bind itself to this event type.
+* Create `App Builder` project(`Runtime`) with webhook using [App builder project and I/O Events](./runtime_webhooks.md) article. **DON'T go to `Developer Console` to create registrations!**
+* Install `aio-cli-plugin-extension` plugins using AIO CLI(`aio plugins discover -i`)
+* Declare your action as `non-web` and set `require-adobe-auth` to `false` in `app.config.yaml` file. I/O Events will be able to call your webhook, but it won't be publicly accessible to anyone else.
```
Comment (sgera) [inline src/pages/guides/autosubscriptions.md:13]: @abhupadh Can you please confirm this?

Comment (sgera) [reply src/pages/guides/autosubscriptions.md:13]: Also, will we get OOTB signature validation in this case if an event registration is created using `aio-cli`?

File: gatsby-config.js [hunk 3]
```diff
@@ -51,6 +51,10 @@ module.exports = {
             title: "Introduction to Journaling",
             path: "guides/journaling_intro.md",
           },
+          {
+            title: "App Builder Auto Subscriptions",
```
Comment (sgera) [inline gatsby-config.js:55]: We have been consistent in our usage of the term "event registration" for I/O Events. I think we should not introduce another term `subscription` for this. 

How about `Automatic event registrations using App Builder`? Or do you have any other suggestion?

File: src/pages/guides/autosubscriptions.md (hunk 2 above)
Comment (kokoc) [reply src/pages/guides/autosubscriptions.md:13]: yes, it will work

File: gatsby-config.js (hunk 3 above)
Comment (kokoc) [reply gatsby-config.js:55]: This is not a part of app builder. I would like to drop `app builder` from the phrase to avoid confusion

Comment (kokoc) [reply gatsby-config.js]: fixed

File: src/pages/guides/autosubscriptions.md (hunk 2 above)
Comment (abhupadh) [reply src/pages/guides/autosubscriptions.md:13]: Yes, they will.

File: gatsby-config.js (hunk 3 above)
Comment (sgera) [reply gatsby-config.js]: Thanks! Can we please remove the `subscriptions` word from the content of `autosubscriptions.md` too? And also rename the file accordingly?

Comment (kokoc) [reply gatsby-config.js]: yes, done

File: src/pages/guides/autoregistrations.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+The integration between [App builder project and I/O Events](./runtime_webhooks.md) allows to create applications that listen to Adobe events. Automatic events registrations push this concept further by subscribing your newly deployed project to `I/O Events` automatically, so you can easily deploy your application in different environments or even share your application with other organizations. Also, this technology minimizes the manual routine work for admins and reduces the possibility to mess up things during manual setup in `Developer Console`.
+
+## Creating self-contained application
+In this chapter, we will create a code that listens to a specific event type and bind itself to this event type.
+* Create `App Builder` project(`Runtime`) with webhook using [App builder project and I/O Events](./runtime_webhooks.md) article. **DON'T go to `Developer Console` to create registrations!**
+* Install `aio-cli-plugin-extension` plugins using AIO CLI(`aio plugins discover -i`)
+* Declare your action as `non-web` and set `require-adobe-auth` to `false` in `app.config.yaml` file. I/O Events will be able to call your webhook, but it won't be publicly accessible to anyone else.
```
Comment (abhupadh) [inline src/pages/guides/autoregistrations.md:13]: A correction here - the webhook created by developer console using your runtime action beneath **will** be publicly accessible. thanks @sgera for pointing out.

**Review by abhupadh — CHANGES_REQUESTED** [review]

File: src/pages/guides/autoregistrations.md (hunk 4 above)
Comment (kokoc) [reply src/pages/guides/autoregistrations.md]: done

**Review by abhupadh — APPROVED** [review]
👍

**Review by sgera — APPROVED** [review]

**Pull request merged**